src/community/wps-jdbc/src/main/java/org/geoserver/wps/jdbc/JDBCStatusStore.java

```diff
@@ -0,0 +1,493 @@
+                attrs.put(p.getName().toString(), p.getValue());
+            }
+        }
+        Name processName = new NameImpl((String) attrs.get(PROCESS_NAME));
+        String executionId = (String) attrs.get(PROCESS_ID);
+        boolean asynchronous = (boolean) attrs.get(ASYNC);
+
+        ExecutionStatus status = new ExecutionStatus(processName, executionId, asynchronous);
+        if (attrs.containsKey(REQUEST)) {
+            ExecuteType request = buildRequest(attrs);
+
+            status.setRequest(request);
+        }
+        String phase = (String) attrs.get(PHASE);
+        ProcessState state = ProcessState.valueOf(phase);
+        status.setPhase(state);
+
+        status.setProgress((float) attrs.get(PROGRESS));
+        status.setTask((String) attrs.get(TASK));
+        status.setUserName((String) attrs.get(USER_NAME));
+        if (attrs.containsKey(EXCEPTION_MESSAGE)) {
+            status.setException(buildException(attrs, status));
+        }
+        // set the time stamps last as other items may set them for us but
+        // we know best here!
+        status.setCreationTime((Date) attrs.get(CREATION));
+        if (attrs.containsKey(COMPLETION)) {
+            status.setCompletionTime((Date) attrs.get(COMPLETION));
+        }
+        if (attrs.containsKey(LASTUPDATE)) {
+            status.setLastUpdated((Date) attrs.get(LASTUPDATE));
+        }
+
+        return status;
+    }
+
+    private ExecuteType buildRequest(HashMap<String, Object> attrs) {
+        String req = (String) attrs.get(REQUEST);
+        BufferedReader reader = new BufferedReader(new StringReader(req));
+        org.geotools.xml.Parser parser = new Parser(new WPSConfiguration());
+        ExecuteType request = null;
+        try {
+            request = (ExecuteType) parser.parse(reader);
+        } catch (IOException | SAXException | ParserConfigurationException e) {
+            // TODO Auto-generated catch block
+            e.printStackTrace();
+        }
+
+        return request;
+    }
+
+    private Exception buildException(HashMap<String, Object> attrs, ExecutionStatus status) {
+        String message = (String) attrs.get(EXCEPTION_MESSAGE);
+        Exception exc = new Exception(message);
+        // see if we can rebuild the exception
+        try {
+            Constructor<?> con = this.getClass().getClassLoader()
+                    .loadClass((String) attrs.get(EXCEPTION_CLASS)).getConstructor(String.class);
+            exc = (Exception) con.newInstance(message);
+
+        } catch (InstantiationException | IllegalAccessException | ClassNotFoundException e) {
+            // too bad, I don't care
+        } catch (NoSuchMethodException e) {
+            // too bad, I don't care
+        } catch (SecurityException e) {
+            // too bad, I don't care
+        } catch (IllegalArgumentException e) {
+            // too bad, I don't care
+        } catch (InvocationTargetException e) {
+            // too bad, I don't care
```

Comment:
Just one catch would have been sufficient I guess? ;-)
Anyways, I understand you don't care, but this will result in some differences/misbehaviors, maybe at least log it at FINE level?